R/coalesce.R

```diff
@@ -1,37 +1,12 @@
 coalesce = function(x, ..., .dots=NULL, .inplace=FALSE) {
   if (is.null(.dots) && missing(..1)) return(x)
-  
+
   values = if (!is.null(.dots)) {
     if (!missing(..1)) stop("provide either '...' or '.dots' argument, not both")
     if (!is.list(.dots)) stop("argument '.dots' requires a list input")
     .dots
   } else list(...)
-  
-  nn = length(x)
-  # can be lengths() in R 3.2.0
-  lengths = vapply_1i(values, length)
-  if (any(idx <- lengths > 1L & lengths < nn)) {
-    first_idx = which(idx)
-    stop("Replacement ", first_idx, " has length ", lengths[first_idx], 
-         " but length(x) is ", lx, ". Only singletons will be recycled")
-  }
-  
-  x_type = typeof(x)
-  x_factor = inherits(x, 'factor')
-  
-  types = vapply_1c(values, typeof)
-  if (!all(idx <- x_type == types)) {
-    first_idx = which(!idx)
-    stop("Replacement ", first_idx, " has internal type ", types[first_idx],
-         " but typeof(x) is '", x_type, "'. Please coerce before coalescing.")
-  }
-  if (x_type == 'integer' && !all(idx <- x_factor == vapply_1b(values, inherits, 'factor'))) {
```

Comment:
not sure if current codes validates factors well, need to add tests

Reply:
open Q is how exactly `coalesce` should work for factors.

```
coalesce(factor(c('a', NA)), factor('b'))
```

what should happen?

Reply:
what about

```
coalesce(factor(c('b', 'c', NA)), factor(c('a', 'a', 'b')))
```

Reply:
I think output can be `factor(c('a', 'b'))` and `factor(c('b', 'c', 'b'))`, but quite tricky to implement in general...

Reply:
you can put that question in coalesce issue, not here in PR, maybe we will get some feedback there

Reply:
maybe also just have very strict rule/minimal support (match all levels of all replacement options) for initial PR and file follow-up later?